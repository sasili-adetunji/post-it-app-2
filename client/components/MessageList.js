import React from 'react';
import { CardHeader, CardTitle, Card } from 'material-ui/Card';
import TextField from 'material-ui/TextField';
import { CircularProgress } from 'material-ui/CircularProgress';
import RaisedButton from 'material-ui/RaisedButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Paper from 'material-ui/Paper';
import PostItActions from '../actions/PostItActions';
import PostItStore from '../stores/PostItStore';

import Message from './Message';
import { List } from 'material-ui/List';


class MessageList extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const messageNodes = this.props.messages.map((message, i) => {
      return (
        <Message message={message} key={i} />
      );
    });

    return (
      <div>
        <List>
          <CardTitle title="Messages" />

          {messageNodes}
        </List>
      </div>
    );
  }
    }


export default MessageList;
