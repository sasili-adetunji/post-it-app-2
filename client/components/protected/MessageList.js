import React from 'react';
import MessageBox from './MessageBox';
import Message from './Message';


/**
 * creates a messagelist components
 *
 * @class MessageList
 * @extends {React.Component}
 */
class MessageList extends React.Component {

  /**
   *
   * renders messagelist components
   * @returns { void }
   * @memberof MessageList
   */
  render() {
    const messageNodes = this.props.messages.map((message, i) => {
      return (
        <Message
          message={message} key={i} MessageId={this.props.messages[0]}
          readUser={this.props.readUsers} />
      );
    });
    return (
      <div>
        <div className=" top-bar">
          <div className="col-md-8 col-xs-8">
            <h3 className="panel-title"><span className="glyphicon glyphicon-comment" />
              Group 1 Message </h3>
          </div>
        </div>
        {messageNodes}
        <div> </div>
        <MessageBox groupId={this.props.selectedGroup[0]} author={this.props.loggedInUser} />
      </div>
    );
  }
}
export default MessageList;
