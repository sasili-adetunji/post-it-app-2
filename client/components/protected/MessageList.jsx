import React from 'react';
import { Modal } from 'react-bootstrap';
import MessageBox from './MessageBox';
import AddMember from './AddMember';
import Message from './Message';
import AppStore from '../../stores/AppStore';

/**
* A collection of message that displays the available messages
* on the message board
*
* @class MessageList
*
* @extends {React.Component}
*/
class MessageList extends React.Component {
/**
* @description Creates an instance of MessageBox.
* bind methods and set initial state.
*
* @memberof MessageBox
*
* @param {object} props
*/
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
    };
    this.closeModal = this.closeModal.bind(this);
    this.openModal = this.openModal.bind(this);
  }

/**
* Handles close Modal event
*
* @param {SyntheticEvent} event
*
* @returns {void} void
*/
  closeModal() {
    this.setState({ isOpen: false });
  }

/**
* Handles Open Modal event
*
* @param {SyntheticEvent} event
*
* @returns {void} void
*/
  openModal(event) {
    event.preventDefault();
    this.setState({ isOpen: true });
  }

/**
* Renders MessageList componet
*
* @returns {ReactElement} MessageList markup
*/
  render() {
    let messageNodes = null;
    let groupName = null;
    if (AppStore.getOpenedGroup().length === 0) {
      messageNodes = (<div> <h2 className="messageHeader"> No Group Selected
         </h2> </div>);
    } else if (AppStore.getGroupsMessages().length === 0) {
      groupName = (<div> <h4> Group | &nbsp;
        {AppStore.getOpenedGroup()[0].groupName}
      </h4> </div>);
      messageNodes = (<div> <h2 className="messageHeader"> No Message in Group
        </h2> </div>);
    } else {
      groupName = (<div> <h4> Group | &nbsp;
        {AppStore.getOpenedGroup()[0].groupName}
      </h4> </div>);
      messageNodes = AppStore.getGroupsMessages().map((message, i) => (
        <Message
          message={message}
          key={i}
          MessageId={AppStore.getGroupsMessages()[0]}
          readUser={AppStore.getReadUsers()}
        />
      ));
    }
    return (
      <div>
        <div className="row">
          <div className="col-md-10">
            {groupName}
          </div>
          <div className="col-md-2">
            <a onClick={this.openModal}> <b> Add new Member </b> </a>
          </div>
        </div>
        <Modal show={this.state.isOpen} onHide={this.closeModal}>
          <Modal.Body>
            <AddMember />
          </Modal.Body>
          <Modal.Footer>
            <a onClick={this.closeModal}> Close</a>
          </Modal.Footer>
        </Modal>
        <div className="messages">
          {messageNodes}
        </div>
        <MessageBox />
      </div>
    );
  }
}
export default MessageList;
