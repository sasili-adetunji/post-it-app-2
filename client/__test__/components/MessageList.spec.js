import React from 'react';
import { shallow, render } from 'enzyme';
import MessageList from '../../components/protected/MessageList.jsx';

describe('MessageList components', () => {
  const group = { groupId: 'TUsUaucS5nb5kuOP6SpvTUcX9v3',
    groupName: 'Test Group' };
  const readUsers = [{ K345SHRKSDFKDSDGjhe5: 'Adeola' }, { K95745SHRKSDFKDSDGjhe5: 'Kola' }];
  const message = [{ messageBody: 'Test message',
    postedBy: 'testUser@gmail.com',
    postedByDisplayName: 'Test User',
    postedon: '',
    priority: 'Normal' }];
  const component = shallow(<MessageList selectedGroup={group} readUsers={readUsers} messages={message} />);
  it('should match snapshot test', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render', () => {
    expect(component).toBeDefined();
  });
  it('Should contain three div', () => {
    expect(component.find('div').length).toEqual(3);
  });
  it('should recieve props', () => {
    expect(Object.keys(component.props()).length).toBeGreaterThan(0);
  });
});
