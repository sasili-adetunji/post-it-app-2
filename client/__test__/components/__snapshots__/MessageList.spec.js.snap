// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList components should match snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="viewMessage"
      id="mesa"
    >
      <h2
        className="messageHeader"
      >
         No Message in Group 
      </h2>
    </div>
    <div
      id="footer"
    >
      <MessageBox
        author={undefined}
        groupId={undefined}
      />
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div
        className="viewMessage"
        id="mesa"
      >
        <h2
          className="messageHeader"
        >
           No Message in Group 
        </h2>
      </div>
      <div
        id="footer"
      >
        <MessageBox
          author={undefined}
          groupId={undefined}
        />
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <MessageList
      messages={
        Array [
          Object {
            "messageBody": "Test message",
            "postedBy": "testUser@gmail.com",
            "postedByDisplayName": "Test User",
            "postedon": "",
            "priority": "Normal",
          },
        ]
      }
      readUsers={
        Array [
          Object {
            "K345SHRKSDFKDSDGjhe5": "Adeola",
          },
          Object {
            "K95745SHRKSDFKDSDGjhe5": "Kola",
          },
        ]
      }
      selectedGroup={
        Object {
          "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
          "groupName": "Test Group",
        }
      }
    />,
    "_instance": MessageList {
      "context": Object {},
      "onChange": [Function],
      "props": Object {
        "messages": Array [
          Object {
            "messageBody": "Test message",
            "postedBy": "testUser@gmail.com",
            "postedByDisplayName": "Test User",
            "postedon": "",
            "priority": "Normal",
          },
        ],
        "readUsers": Array [
          Object {
            "K345SHRKSDFKDSDGjhe5": "Adeola",
          },
          Object {
            "K95745SHRKSDFKDSDGjhe5": "Kola",
          },
        ],
        "selectedGroup": Object {
          "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
          "groupName": "Test Group",
        },
      },
      "refs": Object {},
      "state": Object {
        "message": "",
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div>
      <div
        className="viewMessage"
        id="mesa"
      >
        <h2
          className="messageHeader"
        >
           No Message in Group 
        </h2>
      </div>
      <div
        id="footer"
      >
        <MessageBox
          author={undefined}
          groupId={undefined}
        />
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MessageList
    messages={
      Array [
        Object {
          "messageBody": "Test message",
          "postedBy": "testUser@gmail.com",
          "postedByDisplayName": "Test User",
          "postedon": "",
          "priority": "Normal",
        },
      ]
    }
    readUsers={
      Array [
        Object {
          "K345SHRKSDFKDSDGjhe5": "Adeola",
        },
        Object {
          "K95745SHRKSDFKDSDGjhe5": "Kola",
        },
      ]
    }
    selectedGroup={
      Object {
        "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
        "groupName": "Test Group",
      }
    }
  />,
}
`;
